### ISSUE TITLE
Type could not be found error using assembly and type console arguments

### ISSUE THREAD

**Issue by the-real-paul** [ch-robinson/dotnet-avro#99]
Using the `--assembly` and `--type` arguments on the commandline of the Chr.Avro.Cli tool (e.g. `create --type KafkaTests.Models.TestAvro --assembly C:\Projects\KafkaTests\bin\Debug\netcoreapp3.1\KafkaTests.Models.dll`), I'm getting an error "The type could not be found. You may need to provide additional assemblies."  Looking into the code, I can see that the `ResolveType` extension methods is loading assemblies, but those assemblies do not get saved to any variables, so when the code gets to this line...

`return Type.GetType(options.TypeName, ignoreCase: true, throwOnError: true);`

... it's not actually using the assembly that was loaded from the previous lines.  I'm proposing this be fixed by adding a collection of Assemblies that the `Assembly.Load` or the `Assembly.LoadFrom` methods return and then iterating through that array to find the type that belongs to one of these assemblies.

```
    internal static class TypeOptionExtensions
    {
        public static Type ResolveType(this IClrTypeOptions options)
        {
            List<Assembly> assemblies = new List<Assembly>();
            foreach (var assembly in options.AssemblyNames)
            {
                try
                {
                    // If found, save this assembly to the assemblies collection.
                    assemblies.Add(Assembly.Load(assembly));
                    continue;
                }
                catch (FileNotFoundException)
                {
                    // nbd
                }
                catch (FileLoadException)
                {
                    // also nbd
                }

                try
                {
                    // If found, save this assembly to the assemblies collection.
                    assemblies.Add(Assembly.LoadFrom(Path.GetFullPath(assembly)));
                }
                catch (FileNotFoundException)
                {
                    throw new ProgramException(message: $"{assembly} could not be found. Make sure that you’ve provided either a recognizable name (e.g. System.Runtime) or a valid assembly path.");
                }
                catch (BadImageFormatException)
                {
                    throw new ProgramException(message: $"{assembly} is not valid. Check that the path you’re providing points to a valid assembly file.");
                }
            }

            try
            {
                // Iterate through the loaded assemblies to find the first one that contains the given type.
                foreach (var assembly in assemblies)
                {
                    Type type = assembly.GetType(options.TypeName, throwOnError: false, ignoreCase: true);
                    if (type != null)
                    {
                        return type;
                    }
                }

                return Type.GetType(options.TypeName, throwOnError: true, ignoreCase: true);
            }
            catch (TypeLoadException)
            {
                throw new ProgramException(message: "The type could not be found. You may need to provide additional assemblies.");
            }
        }
    }
```

**Comment by dstelljes** [comment]
Thanks for the report! This looks like it was introduced with #96.

I just opened #100 to address this—it’s more or less the fix you suggested, but with some additional resolution logic to ensure that assemblies with dependencies are handled properly.

**Comment by dstelljes** [comment]
Fixed with 7.0.2.
